File: assignments/assignment-1-solution/src/CrazyEights/Cards/Card.cs
using CrazyEights.Domain;

namespace CrazyEights.Cards;

public class Card : ICard
{
    public Card(RankType rankType, SuitType suitType)
    {
        Rank = rankType;
        Suit = suitType;
    }

    public string GetDescription()
    {
        var rankName = Domain.Rank.GetRankName(Rank);
        var suitName = Domain.Suit.GetSuitName(Suit);
        return $"{rankName} of {suitName}";
    }

    public bool IsSelectable => true;

    public RankType Rank { get; }

    public SuitType Suit { get; }

    public int CardId => (int)Suit * 100 + (int)Rank;

    public override string ToString()
    {
        return GetDescription();
    }
}

File: assignments/assignment-1-solution/src-java/src/main/java/edu/kennesaw/crazy8s/cards/Card.java
package edu.kennesaw.crazy8s.cards;

import edu.kennesaw.crazy8s.domain.RankType;
import edu.kennesaw.crazy8s.domain.SuitType;

/**
 * Defines the core properties and behaviors of a card.
 */
public interface Card {
    /**
     * Gets the rank of the card.
     */
    RankType getRank();

    /**
     * Gets the suit of the card.
     */
    SuitType getSuit();

    /**
     * Gets a unique identifier for the card.
     */
    int getCardId();

    /**
     * Gets a value indicating whether the card can be selected to play.
     */
    boolean isSelectable();

    /**
     * Returns a human-readable description of the card.
     */
    String getDescription();
}
